4-projects/business_unit_1/development/example_storage_cmek.tf

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "env_secrets_project" {
+  source                      = "../../modules/single_project"
+  impersonate_service_account = var.terraform_service_account
+  org_id                      = var.org_id
+  billing_account             = var.billing_account
+  folder_id                   = data.google_active_folder.env.name
+  environment                 = "development"
+  alert_spent_percents        = var.alert_spent_percents
+  alert_pubsub_topic          = var.alert_pubsub_topic
+  budget_amount               = var.budget_amount
+  project_prefix              = "sample-env-secrets"
+
+  activate_apis = ["logging.googleapis.com", "secretmanager.googleapis.com", "cloudkms.googleapis.com"]
+
+  # Metadata
+  application_name  = "bu1-sample-application"
+  billing_code      = "1234"
+  primary_contact   = "example@example.com"
+  secondary_contact = "example2@example.com"
+  business_code     = "bu1"
+}
+
+module "kms" {
+  source  = "terraform-google-modules/kms/google"
+  version = "~> 0.1"
+
+  project_id = module.env_secrets_project.project_id
+  keyring    = "sample-keyring"
+  location   = "global"
+  keys       = ["crypto-key-example"]
+}
+
+data "google_storage_project_service_account" "gcs_account" {
+}
+
+data "google_iam_policy" "admin" {
+  binding {
+    role = "roles/cloudkms.cryptoKeyEncrypter"
+
+    members = [
+      "serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}"
+    ]
+  }
+}
+
+resource "google_kms_crypto_key_iam_policy" "crypto_key" {
+  crypto_key_id = module.kms.keys
+  policy_data   = data.google_iam_policy.admin.policy_data
+}
+
+
+resource "google_storage_bucket" "bucket" {
+  name    = "cmek-encrypted-bucket"
+  project = module.base_shared_vpc_project.project_id
```

Comment:
Shouldn't the bucket be in `module.env_secrets_project.project_id`

Reply:
guidance from Pete's requirement doc was to keep the keyring and the GCS bucket in different projects to maintain separation of secret from the workload. Let me know if this understanding has changed or you want me to exchange out the project location for another one or even a variable that the user can provide input to.

Reply:
That makes sense, thanks for the clarification!